src/api/app/assets/javascripts/webui2/project_monitor.js

```diff
@@ -0,0 +1,113 @@
+function hideRepositoryColumns() {
+  var table = $('#project-monitor-table').DataTable();
+
+  var repositories = [];
+  $('#project-monitor-repositories-dropdown input:checked').each(function () {
+    repositories.push($(this).val().trim());
+  });
+
+  if (repositories.length === 0) {
+    return;
+  }
+
+  var toShowColumns = [];
+  var offSet = 0;
+  $('#project-monitor-table thead tr:eq(0) th').each(function () {
+    var column = $(this);
+    var colSpan = column.prop('colspan');
+    if (repositories.includes(column.text().trim())) {
+      var range = [];
+      for (var i = offSet; i < offSet + colSpan; i++) {
+        range.push(i);
+      }
+      toShowColumns = toShowColumns.concat(range);
+    }
+    offSet += colSpan;
+  });
+
+  table.columns().every(function () {
+    var index = this.index();
+    if (index === 0) return;
+    if (toShowColumns.includes(index)) {
```

Comment:
The `if/else` can be refactored in `this.visible(toShowColumns.includes(index));`